truss/remote/baseten/remote.py

```diff
@@ -175,44 +164,167 @@ def push(  # type: ignore
                 "Deployment name must only contain alphanumeric, -, _ and . characters"
             )
 
+        model_id = exists_model(self._api, model_name)
+
         if model_id is not None and disable_truss_download:
             raise ValueError("disable-truss-download can only be used for new models")
 
+        temp_file = archive_truss(gathered_truss)
+        s3_key = upload_truss(self._api, temp_file)
         encoded_config_str = base64_encoded_json_str(
             gathered_truss._spec._config.to_dict()
         )
 
-        temp_file = archive_truss(gathered_truss)
-        s3_key = upload_truss(self._api, temp_file)
-
-        model_id, model_version_id = create_truss_service(
-            api=self._api,
+        return FinalPushData(
             model_name=model_name,
             s3_key=s3_key,
-            config=encoded_config_str,
+            encoded_config_str=encoded_config_str,
             is_draft=not publish,
             model_id=model_id,
             is_trusted=trusted,
             preserve_previous_prod_deployment=preserve_previous_prod_deployment,
-            deployment_name=deployment_name,
+            version_name=deployment_name,
             origin=origin,
             environment=environment,
-            chain_environment=chain_environment,
-            chainlet_name=chainlet_name,
-            chain_name=chain_name,
             allow_truss_download=not disable_truss_download,
         )
 
+    def push(  # type: ignore
+        self,
+        truss_handle: TrussHandle,
+        model_name: str,
+        publish: bool = True,
+        trusted: bool = False,
+        promote: bool = False,
+        preserve_previous_prod_deployment: bool = False,
+        disable_truss_download: bool = False,
+        deployment_name: Optional[str] = None,
+        origin: Optional[custom_types.ModelOrigin] = None,
+        environment: Optional[str] = None,
+    ) -> BasetenService:
+        push_data = self._prepare_push(
+            truss_handle=truss_handle,
+            model_name=model_name,
+            publish=publish,
+            trusted=trusted,
+            promote=promote,
+            preserve_previous_prod_deployment=preserve_previous_prod_deployment,
+            disable_truss_download=disable_truss_download,
+            deployment_name=deployment_name,
+            origin=origin,
+            environment=environment,
+        )
+
+        model_id, model_version_id = create_truss_service(
```

Comment:
Any reason not to pass `push_data` as an argument? What is the intended abstraction here?

Reply:
If you don't wanna address it, could you explain why or leave a TODO?

Reply:
Sorry I forgot to respond to this comment. I think I'm generally not happy with how often these arguments are duplicated across functions (with default values on many args). I can put a `TODO` to consolidate all of these occurrences.